### PR TITLE
Create simple Inception model architecture

### PR DESCRIPTION
**Pull request by alisha17**
**Results for Inception model:**

Epoch 1/50
844/843 [==============================] - 1131s 1s/step - loss: 0.2375 - accuracy: 0.6845
Epoch 2/50
844/843 [==============================] - 1342s 2s/step - loss: 0.2132 - accuracy: 0.7207
Epoch 3/50
844/843 [==============================] - 1259s 1s/step - loss: 0.2010 - accuracy: 0.7272
Epoch 4/50
844/843 [==============================] - 1412s 2s/step - loss: 0.1997 - accuracy: 0.7219
Epoch 5/50
844/843 [==============================] - 1253s 1s/step - loss: 0.1969 - accuracy: 0.7279
Epoch 6/50
844/843 [==============================] - 1168s 1s/step - loss: 0.1871 - accuracy: 0.7444
Epoch 7/50
844/843 [==============================] - 1221s 1s/step - loss: 0.2031 - accuracy: 0.7195
'
'
'
Epoch 25/50
844/843 [==============================] - 1241s 1s/step - loss: 0.1515 - accuracy: 0.7904
Epoch 26/50
844/843 [==============================] - 1157s 1s/step - loss: 0.1432 - accuracy: 0.8038
Epoch 27/50
844/843 [==============================] - 1227s 1s/step - loss: 0.1513 - accuracy: 0.7942
Epoch 28/50
844/843 [==============================] - 1223s 1s/step - loss: 0.1387 - accuracy: 0.8103
Epoch 29/50
844/843 [==============================] - 1135s 1s/step - loss: 0.1460 - accuracy: 0.7998
Epoch 30/50
844/843 [==============================] - 1143s 1s/step - loss: 0.1456 - accuracy: 0.8031

The model shows quite better accuracy than the VGG16 model. The loss function is also decreasing after every epoch. The model might have started overfitting after 28th epoch. However, we can't be so sure as we need to run for more epochs to say that confidently.

**Results for VGG16 model:**

Epoch 1/50
879/878 [==============================] - 138s 157ms/step - loss: 0.2500 - accuracy: 0.5508
Epoch 2/50
879/878 [==============================] - 139s 158ms/step - loss: 0.2391 - accuracy: 0.6031
Epoch 3/50
879/878 [==============================] - 138s 157ms/step - loss: 0.2327 - accuracy: 0.6231
Epoch 4/50
879/878 [==============================] - 142s 162ms/step - loss: 0.2335 - accuracy: 0.6185
Epoch 5/50
879/878 [==============================] - 139s 158ms/step - loss: 0.2344 - accuracy: 0.6095
Epoch 6/50
879/878 [==============================] - 148s 169ms/step - loss: 0.2274 - accuracy: 0.6310
Epoch 7/50
879/878 [==============================] - 153s 174ms/step - loss: 0.2232 - accuracy: 0.6350
Epoch 8/50
879/878 [==============================] - 156s 177ms/step - loss: 0.2165 - accuracy: 0.6560

The loss function is not decreasing after every epoch very significantly. So, overall the Inception model performs better than the VGG16 model (atleast in this basic setup).

### CODE REVIEW

**Comment by sagarvijaygupta** [comment]
Can you share the confusion matrix in both cases? Dataset is imbalanced by a good margin for 'n'. So just marking all images in 'y' will itself give more than 55% accuracy and a good judge between 'y' and 'd' only excluding 'n' can also reach upto 90%. I don't think accuracy alone can do the justification, confusion matrix will give a much better insight into this. (271-n, 1667-y, 1110-d)

**Comment by Zzoey** [comment]
I was wondering the same. The class imbalance needs to be considered, therefore some sampling might be essential.

**Comment by marco-c** [comment]
In the pretrain.py script, the dataset should already be balanced (https://github.com/marco-c/autowebcompat/blob/576ca42d721fc6df890aeb7a9ff25057a5f1863f/pretrain.py#L51).

File: network.py [hunk 1]
```diff
@@ -88,6 +88,31 @@ def create_vgglike_network(input_shape):
...
+
+    x2 = Conv2D(64, (1, 1), activation='relu', padding='same')(input)
+    x2 = Conv2D(64, (5, 5), activation='relu', padding='same')(x2)
+
+    x3 = MaxPooling2D((3, 3), strides=(1, 1), padding='same')(input)
+    x3 = Conv2D(64, (1, 1), activation='relu', padding='same')(x3)
+
+    x = concatenate([x1, x2, x3], axis=3)
+    x = Flatten()(x)
+    x = Dense(2, activation='softmax')(x)
```
Comment (marco-c) [inline network.py]: We probably don't need the softmax layer here because of the architecture we're using.
It might be worth retesting with a Dense layer instead (like in the `vgg16` or `vgglike` cases).

**Comment by alisha17** [comment]
The performance of Inception vs VGG16

Inception:

Epoch 1/50
843/842 [==============================] - 6468s 8s/step - loss: 0.2548 - accuracy: 0.4838
Epoch 2/50
843/842 [==============================] - 6904s 8s/step - loss: 0.2501 - accuracy: 0.4966
Epoch 3/50
843/842 [==============================] - 6030s 7s/step - loss: 0.2493 - accuracy: 0.5329
Epoch 4/50
843/842 [==============================] - 5969s 7s/step - loss: 0.2472 - accuracy: 0.5914
Epoch 5/50
843/842 [==============================] - 5949s 7s/step - loss: 0.2392 - accuracy: 0.6207
Epoch 6/50
843/842 [==============================] - 6053s 7s/step - loss: 0.2298 - accuracy: 0.6341
Epoch 7/50
843/842 [==============================] - 6026s 7s/step - loss: 0.2278 - accuracy: 0.6327
Epoch 8/50
843/842 [==============================] - 5985s 7s/step - loss: 0.2195 - accuracy: 0.6588
Epoch 9/50
843/842 [==============================] - 6054s 7s/step - loss: 0.2234 - accuracy: 0.6456
Epoch 10/50
843/842 [==============================] - 6044s 7s/step - loss: 0.2227 - accuracy: 0.6401
Epoch 11/50
843/842 [==============================] - 6041s 7s/step - loss: 0.2196 - accuracy: 0.6428
Epoch 12/50
843/842 [==============================] - 6031s 7s/step - loss: 0.2198 - accuracy: 0.6449
Epoch 13/50
843/842 [==============================] - 6015s 7s/step - loss: 0.2175 - accuracy: 0.6489
Epoch 14/50
843/842 [==============================] - 6003s 7s/step - loss: 0.2082 - accuracy: 0.6735
Epoch 15/50
843/842 [==============================] - 6001s 7s/step - loss: 0.2125 - accuracy: 0.6583

VGG16:

Epoch 1/50
842/841 [==============================] - 1595s 2s/step - loss: 0.2578 - accuracy: 0.5785
Epoch 2/50
842/841 [==============================] - 1484s 2s/step - loss: 0.2535 - accuracy: 0.5961
Epoch 3/50
842/841 [==============================] - 1381s 2s/step - loss: 0.2478 - accuracy: 0.6094
Epoch 4/50
842/841 [==============================] - 1393s 2s/step - loss: 0.2298 - accuracy: 0.6398
Epoch 5/50
842/841 [==============================] - 1402s 2s/step - loss: 0.2156 - accuracy: 0.6662
Epoch 6/50
842/841 [==============================] - 1403s 2s/step - loss: 0.2004 - accuracy: 0.6853
Epoch 7/50
842/841 [==============================] - 1401s 2s/step - loss: 0.1882 - accuracy: 0.7133
Epoch 8/50
842/841 [==============================] - 1408s 2s/step - loss: 0.1768 - accuracy: 0.7335
Epoch 9/50
842/841 [==============================] - 1426s 2s/step - loss: 0.1681 - accuracy: 0.7527
Epoch 10/50
842/841 [==============================] - 1411s 2s/step - loss: 0.1579 - accuracy: 0.7631
Epoch 11/50
842/841 [==============================] - 1409s 2s/step - loss: 0.1487 - accuracy: 0.7866
Epoch 12/50
842/841 [==============================] - 1402s 2s/step - loss: 0.1546 - accuracy: 0.7763
Epoch 13/50
842/841 [==============================] - 1399s 2s/step - loss: 0.1347 - accuracy: 0.8106
Epoch 14/50
842/841 [==============================] - 1397s 2s/step - loss: 0.1309 - accuracy: 0.8120
Epoch 15/50
842/841 [==============================] - 1427s 2s/step - loss: 0.1445 - accuracy: 0.7935

VGG16 performs better than Inception in this case. However, it might depend on different data.

**Comment by marco-c** [comment]
@alisha17 there are some conflicts in network.py that should be resolved before merging this.

**Comment by alisha17** [comment]
@marco-c I rebased on the master

**Comment by alisha17** [comment]
Hi @marco-c , can you explain a bit about the coverage issue? Why does merging this decrease the coverage?

**Comment by marco-c** [comment]
It decreases the coverage because you are adding code that is not tested, but it's fine, we can add tests in the future.
